lib/networkx/graph.rb

```diff
@@ -117,5 +117,216 @@ def remove_nodes(nodes)
+    #
+    # @param edges [Array<Object, Object>] the array of edges
+    # @param weight [Array<Integer>] the array of weights
+    def add_weighted_edges(edges, weights)
+      raise ArgumentError, 'edges and weights array must have equal number of elements.'\
+                           unless edges.size == weights.size
+      raise ArgumentError, 'edges and weight must be given in an Array.'\
+                           unless edges.is_a?(Array) && weights.is_a?(Array)
+      (edges.transpose << weights).transpose.each do |node_1, node_2, weight|
+        add_weighted_edge(node_1, node_2, weight)
+      end
+    end
+
+    # Clears the graph
+    #
+    # @example
+    #   graph.clear
+    def clear
+      @adj.clear
+      @nodes.clear
+      @graph.clear
+    end
+
+    # Checks if a node is present in the graph
+    #
+    # @example
+    #   graph.node?(node_1)
+    #
+    # @param node [Object] the node to be checked
+    def node?(node)
+      @node.key?(node)
+    end
+
+    # Checks if the the edge consisting of two nodes is present in the graph
+    #
+    # @example
+    #   graph.edge?(node_1, node_2)
+    #
+    # @param node_1 [Object] the first node of the edge to be checked
+    # @param node_2 [Object] the second node of the edge to be checked
+    def edge?(node_1, node_2)
+      return true if @nodes.key?(node_1) && @adj[node_1].key?(node_2)
+      false
+    end
+
+    # Gets the node data
+    #
+    # @example
+    #   graph.get_node_data(node)
+    #
+    # @param node [Object] the node whose data is to be fetched
+    def get_node_data(node)
+      return @nodes[node] if @nodes.key?(node)
+      raise ArgumentError, 'No such node exists!'
+    end
+
+    # Gets the edge data
+    #
+    # @example
+    #   graph.get_edge_data(node_1, node_2)
+    #
+    # @param node_1 [Object] the first node of the edge
+    # @param node_2 [Object] the second node of the edge
+    def get_edge_data(node_1, node_2)
+      return @adj[node_1][node_2] if @nodes.key?(node_1) && @adj[node_1].key?(node_2)
+      raise KeyError, 'No such edge exists!'
+    end
+
+    # Retus a hash of neighbours of a node
+    #
+    # @example
+    #   graph.neighbours(node)
+    #
+    # @param node [Object] the node whose neighbours are to be fetched
+    def neighbours(node)
+      return @adj[node] if @nodes.key?(node)
+      raise KeyError, 'No such node exists!'
+    end
+
+    # Returns number of nodes
+    #
+    # @example
+    #   graph.number_of_nodes
+    def number_of_nodes
+      @nodes.length
+    end
+
+    # Returns number of edges
+    #
+    # @example
+    #   graph.number_of_edges
+    def number_of_edges
+      num = 0
+      @adj.each { |_, v| num += v.length }
+      num / 2
+    end
+
+    # Returns number of edges if is_weighted is false
+    # or returns total weight of all edges
+    #
+    # @example
+    #   graph.size(true)
+    #
+    # @praram is_weighted [Bool] if we want weighted size of unweighted size
```

Comment:
Make same documentation comment changes, as mentioned to the same method in the previous file. (Logic explanation moved to params + typo)